src/connection.rs

```diff
@@ -196,9 +219,22 @@ impl Connection {
             incoming_throughput,
             reconnects,
         );
+
+        for (pkid, ack_time)  in  pub_acks.into_iter() {
+            let publish_time = pkids_publish.get(&pkid).unwrap();
+            let latency = publish_time.duration_since(ack_time).as_millis();
```

Comment:
This is a difference between incoming ack and incoming publish. What we want is difference between `outgoing publish`  and `incoming ack` for publish latencies (or) `outgoing publish` and `incoming publish` for round trip latencies

Reply:
Ah my bad. 

One thing though, the `Outgoing` publish is value is `Batch`. How can we get `pkids` of an internally batched payload?

https://docs.rs/rumqttc/0.0.6/rumqttc/enum.Outgoing.html